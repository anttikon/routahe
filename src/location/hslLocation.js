import fetch from 'node-fetch'
import {head} from 'lodash'

async function getHslLocations(query) {
  const response = await fetch(getUrl('http://api.digitransit.fi/geocoding/v1/search', {
    'text': encodeURIComponent(query),
    'boundary.rect.min_lat': 59.9,
    'boundary.rect.max_lat': 60.45,
    'boundary.rect.min_lon': 24.3,
    'boundary.rect.max_lon': 25.5
  }))
  return response.json()
}

async function getLocationByQuery(query) {
  const hslLocations = await getHslLocations(query)
  if (!hslLocations.features) {
    return {}
  }
  return head(hslLocations.features.map(mapFeature))

  function mapFeature(feature) {
    const [lon, lat] = feature.geometry.coordinates
    const label = feature.properties.label
    return {label, lon, lat}
  }
}

function getUrl(api, params) {
  const urlParams = Object.keys(params).map(key => `${key}=${params[key]}`).join('&')
  return `${api}?${urlParams}`
}


export {getHslLocations, getLocationByQuery}