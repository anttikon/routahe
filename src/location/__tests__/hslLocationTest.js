import {getHslLocations} from '../hslLocation'

describe('location', function() {
  jasmine.DEFAULT_TIMEOUT_INTERVAL = 50000
  it('should fetch location by wifi if query is empty', async() => {
    const locations = await getHslLocations('Helsingin päärautatieasema')
    expect(locations.features).toMatchSnapshot()
  })

  it('should return empty array with empyt query', async() => {
    const locations = await getHslLocations('')
    expect(locations.features).toEqual([])
  })
})