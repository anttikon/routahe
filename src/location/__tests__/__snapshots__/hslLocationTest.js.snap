exports[`location should fetch location by wifi if query is empty 1`] = `
Array [
  Object {
    "bbox": Array [
      24.940331,
      60.170712,
      24.943317,
      60.172866,
    ],
    "geometry": Object {
      "coordinates": Array [
        24.941498,
        60.171457,
      ],
      "type": "Point",
    },
    "properties": Object {
      "accuracy": "point",
      "confidence": 1,
      "country": "Suomi",
      "country_a": "FIN",
      "country_gid": "whosonfirst:country:85633143",
      "gid": "openstreetmap:venue:way:122595198",
      "id": "way:122595198",
      "label": "Helsingin p채채rautatieasema, Helsinki",
      "layer": "venue",
      "localadmin": "Helsinki",
      "localadmin_gid": "whosonfirst:localadmin:907199715",
      "locality": "Helsinki",
      "locality_gid": "whosonfirst:locality:101748417",
      "name": "Helsingin p채채rautatieasema",
      "neighbourhood": "Kluuvi",
      "neighbourhood_gid": "whosonfirst:neighbourhood:85898847",
      "postalcode": "00100",
      "region": "Uusimaa",
      "region_gid": "whosonfirst:region:85683067",
      "source": "openstreetmap",
      "source_id": "way:122595198",
    },
    "type": "Feature",
  },
]
`;
